src/taipy/core/_entity/_entity.py

```diff
@@ -34,6 +34,6 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
             self._properties.data.update(self._properties._pending_changes)
         _get_manager(self._MANAGER_NAME)._set(self)
 
-        while self._in_context_attributes_changed_collector:
-            _publish_event(*self._in_context_attributes_changed_collector.pop(0))
+        for event in self._in_context_attributes_changed_collector:
+            Notifier.publish(event)
```

Comment:
*Note to reviewers*: Adding more arguments to the `Event` class make the use of a arg list a bit cumbersome and very error prone (think the order being significant!).
That is why I propose to replace this list of args with an `Event` object. Then, the logic does not change, except that it makes more sense to call `Notifier.publish` directly now...

Reply:
that's awesome!!